src/envoy/http/service_control/client_cache.cc

```diff
@@ -320,7 +320,12 @@ CancelFunc ClientCache::callCheck(
                       "due to network_fail_open policy.");
             on_done(Status::OK, response_info);
           } else {
-            on_done(status, response_info);
+            // This is not a client request error, so translate non-5xx error
+            // codes to 500 Internal Server Error. Error message contains
+            // details on the original error (including the original HTTP status
+            // code).
+            Status converted(Code::INTERNAL, status.error_message());
```

Comment:
I am doing similar changes in v1.  I discover that
 for 5xx error, it is better to preserve it.  only convert 4xx errors to INTERNAL
5xx errors will go here if it is network_fail_close.

Reply:
Good point, fixed. Seems like the tests do catch this case.